### PR TITLE
Use snapshots for in-memory copies of layered worldstate

### PR DESCRIPTION
**Pull request by garyschulte**
## PR description

In `BonsaiLayeredWorldState.copy()`, use a snapshot worldstate as the basis for a BonsaiInMemoryWorldstate copy

Prevents stateroot mismatches that arise from the persisted worldstate being different than the worldstate assoicated with a given layered worldstate.

Also moves isolation test setup into an abstract class, and adds a new BonsaiInMemoryIsolationTest

## Documentation

- [ ] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Comment by garyschulte** [comment]
This has been successfully burning in on a goerli node that was previously quite troubled.  In-Memory worldstate performance has been significantly faster than just using snapshots alone:

Screenshot from 2022-11-18 08-38-54

Heap consumption of the in-memory bonsai has been consistently higher, but the total of memory consumed of both heap and are quite close

File: ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiSnapshotWorldStateKeyValueStorage.java [hunk 1]
```diff
@@ -67,6 +67,14 @@ public BonsaiUpdater updater() {
         trieLogStorage);
   }
 
+  @Override
+  public void close() throws Exception {
+    accountStorage.close();
```
Comment (matkt) [inline ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiSnapshotWorldStateKeyValueStorage.java:72]: we don't close trieLogStorage ?

Comment (garyschulte) [reply ethereum/core/src/main/java/org/hyperledger/besu/ethereum/bonsai/BonsaiSnapshotWorldStateKeyValueStorage.java:72]: trie log storage is not based on a snapshot, so we don't have to worry about closing it.

**Review by matkt — APPROVED** [review]
LGTM

**Pull request merged**